pkg/reconciler/openshift/tektonpipeline/extension.go

```diff
@@ -66,39 +69,50 @@ type openshiftExtension struct {
 }
 
 func (oe openshiftExtension) Transformers(comp v1alpha1.TektonComponent) []mf.Transformer {
-	return []mf.Transformer{
-		common.InjectLabelOnNamespace(monitoringLabel),
+	trns := []mf.Transformer{
 		occommon.ApplyCABundles,
 		occommon.RemoveRunAsUser(),
 	}
+
+	pipeline := comp.(*v1alpha1.TektonPipeline)
+
+	// Add monitoring label if metrics is enabled
+	value := findParam(pipeline.Spec.Params, enableMetricsKey)
+	if value == "" || value == "true" {
+		trns = append(trns, common.InjectLabelOnNamespace(monitoringLabel))
+	}
```

Comment:
I think this logic should be handled in the Pre-Reconcile. I think the `openshift-pipelines` namespace is a special case. 
 As it is not created by the TektonPipeline Reconciler, but by TektonPipeline OpenShift Extension (Prereconcile). This is because the targetnamespace is pre-requisite for other things this pre-recociler does.

I could be wrong. However, making this comment to make sure that we have considered the fact that the target-namespace is created by prereconcile.

Reply:
main reconciler has release yaml which also has namespace so that will override on this
but extension has to be refactored as currently it will only install resources in openshift-pipelines namespace.